homeassistant/components/microbot_push/__init__.py

```diff
@@ -0,0 +1,176 @@
+        depth = call.data["depth"]
+        duration = call.data["duration"]
+        mode = call.data["mode"]
+        await coordinator.api.connect()
+        coordinator.api.setDepth(depth)
+        coordinator.api.setDuration(duration)
+        coordinator.api.setMode(mode)
+        await coordinator.api.calibrate()
+        await coordinator.api.disconnect()
+
+    hass.services.async_register(DOMAIN, "generate_token", generate_token)
+    hass.services.async_register(DOMAIN, "calibrate", calibrate)
+    entry.async_on_unload(entry.add_update_listener(async_reload_entry))
+    return True
+
+
+class MicroBotDataUpdateCoordinator(PassiveBluetoothDataUpdateCoordinator):
+    """Class to manage fetching data from the MicroBot."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        client: MicroBotApiClient,
+        ble_device: BLEDevice,
+    ) -> None:
+        """Initialize."""
+        self.api = client
+        self.platforms: list[str] = []
+        self._ready_event = asyncio.Event()
+        self.data: dict[str, Any] = {}
+        self.ble_device = ble_device
+
+        super().__init__(
+            hass, _LOGGER, ble_device.address, bluetooth.BluetoothScanningMode.ACTIVE
+        )
+
+    @callback
+    def _async_handle_bluetooth_event(
+        self,
+        service_info: bluetooth.BluetoothServiceInfoBleak,
+        change: bluetooth.BluetoothChange,
+    ) -> None:
+        """Handle a Bluetooth event."""
+        super()._async_handle_bluetooth_event(service_info, change)
+        if adv := parse_advertisement_data(
+            service_info.device, service_info.advertisement
+        ):
+            self.data = adv.data
+            if self.data:
+                self._ready_event.set()
+            _LOGGER.debug("%s: MicroBot data: %s", self.ble_device.address, self.data)
+            self.api.update_from_advertisement(adv)
+        self.async_update_listeners()
+
+    async def async_wait_ready(self) -> bool:
+        """Wait for the device to be ready."""
+        try:
+            await asyncio.wait_for(self._ready_event.wait(), timeout=55)
+        except asyncio.TimeoutError:
+            return False
+        return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Handle removal of an entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    unloaded = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, platform)
+                for platform in PLATFORMS
+                if platform in coordinator.platforms
+            ]
+        )
+    )
+    if unloaded:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unloaded
+
+
+async def async_reload_entry(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Reload config entry."""
+    await async_unload_entry(hass, entry)
+    await async_setup_entry(hass, entry)
```

Comment:
There is no need to implement this.  A reload will already unload and set it up again
```suggestion
```